src/persist/src/file.rs

```diff
@@ -313,8 +300,22 @@ impl FileBlob {
     }
 }
 
+/// Implementation of [Blob] backed by files.
+#[derive(Debug)]
+pub struct FileBlob {
+    read: FileBlobRead,
```

Comment:
this pattern is a little weird. another option would be to pull out a non-pub FileBlobCore and have both of them call into it. plus certainly other options I'm not thinking of. thoughts?

Reply:
done!